recipes/libtorrent/all/conanfile.py

```diff
@@ -62,65 +81,55 @@ def configure(self):
     def layout(self):
         cmake_layout(self, src_folder="src")
 
-    def _check_compiler_supports_cxx14(self):
-        min_compiler_version = {
-            "Visual Studio": "15",
-            "msvc": "191",
-            "gcc": "5" if Version(self.version) < "2.0.8" else "6",
-            "clang": "5",
-            "apple-clang": "5",
-        }.get(str(self.settings.compiler))
-        if min_compiler_version is None:
-            self.output.warn("Unknown compiler. Assuming it is supporting c++14")
-        if Version(self.settings.compiler.version) < min_compiler_version:
-            raise ConanInvalidConfiguration("This compiler (version) does not support c++ 14.")
-        return True, None
-
-    def validate(self):
-        if Version(self.version) < "2.0":
-            if self.settings.compiler.get_safe("cppstd"):
-                check_min_cppstd(self, 11)
-        else:
-            self._check_compiler_supports_cxx14()
-            if self.settings.compiler.get_safe("cppstd"):
-                check_min_cppstd(self, 14)
-
-        if Version(self.dependencies["boost"].ref.version) < "1.69.0" and \
-           (self.dependencies["boost"].options.header_only or self.dependencies["boost"].options.without_system):
-            raise ConanInvalidConfiguration(f"{self.ref} requires boost with system, which is non-header only in boost < 1.69.0")
-
     def requirements(self):
         # libtorrent 2.0.x [x<=6] have issue for recent boost https://github.com/arvidn/libtorrent/discussions/6757
-        if Version(self.version) < "2.0.0" or "2.0.6" < Version(self.version):
-            self.requires("boost/1.81.0")
+        if Version(self.version) < "2.0.0" or Version(self.version) >= "2.0.7":
+            self.requires("boost/1.81.0", transitive_headers=True)
         else:
-            self.requires("boost/1.76.0")
+            self.requires("boost/1.76.0", transitive_headers=True)
         if self.options.enable_encryption:
-            self.requires("openssl/1.1.1s")
+            self.requires("openssl/1.1.1t", transitive_headers=True, transitive_libs=True)
         if self.options.enable_iconv:
             self.requires("libiconv/1.17")
 
```

Comment:
Should this have Transitive libs for v2?

Reply:
I don't know. Do you have a proof that openssl headers are included by libtorrent public headers?

Reply:
libtorrent/1.2.3 (test package): RUN: ./test_package magnet:?xt=urn:btih:c12fe1c06bba254a9dc9f519b335aa7c1367a88a
dyld: Symbol not found: _iconv
  Referenced from: /usr/lib/libarchive.2.dylib
  Expected in: /Users/jenkins/w/prod-v2/BuildSingleReference/p/libic50650dc30c43d/p/lib/libiconv.2.dylib
 in /usr/lib/libarchive.2.dylib
/bin/sh: line 1: 30073 Abort trap: 6           ./test_package magnet:?xt=urn:btih:c12fe1c06bba254a9dc9f519b335aa7c1367a88a

The v2 failure above says it needed the libs?

Reply:
Why a runtime loader error related to libarchive and system iconv would be related to transitive_headers & transitive_libs of openssl which are compile & link time considerations (moreover of a different lib)?

Reply:
I didn't comment on the openssl line 🤔 I comment on a blank line lol I meant to comment on line 93 not 94